repair/repair.cc

```diff
@@ -1979,6 +1981,39 @@ future<> repair_service::do_rebuild_replace_with_repair(locator::token_metadata_
                 rs.get_metrics().replace_finished_ranges = 0;
                 rs.get_metrics().replace_total_ranges = nr_ranges_total;
             }).get();
+        } else {
+            on_internal_error(rlogger, format("do_rebuild_replace_with_repair: unsupported reason={}", reason));
+        }
+        std::unordered_map<sstring, std::unordered_set<gms::inet_address>> live_nodes_per_dc;
+        std::unordered_map<sstring, size_t> lost_nodes_per_dc;
+        if (source_dc) {
+            if (!topology.get_datacenters().contains(*source_dc)) {
+                throw std::runtime_error(format("{}: Could not find source_dc={} in datacenters={}", op, *source_dc, topology.get_datacenters()));
+            }
+            if (topology.get_datacenters().size() == 1) {
+                rlogger.info("{}: source_dc={} ignored since the cluster has a single datacenter", op, *source_dc);
+                source_dc.reset();
+            } else {
+                topology.for_each_node([&] (const locator::node* node) {
+                    if (node->is_this_node()) {
+                        // Count the rebuilt node as lost.
+                        // For replace, the replaced node must be present in ignore_nodes.
```

Comment:
Why? Ignore_nodes are supposed to contain nodes that are specified by user nodes to ignore during replace. will the node to be ignored added to ignore_nodes automatically?

Reply:
Apparently it is added there.
I don't remember exactly where off the top of my head, but I saw it there (and was surprised by it).

Reply:
The replaced node is NORMAL and DOWN. It is just like a typical node that is down. When the user decided to replace it, it makes sense to exclude it and prevents it from joining the cluster again. 

I am not sure when _topology_state_machine._topology.ignored_nodes are added.

Reply:
In any case, we need to count it as a single lost node in the DC.
My concern is replace using a different ip address that will count both the replaced node and the new node as "lost", while only the replaced node is lost.

Reply:
done in v8